### PR TITLE
Feature/M0078DDG-375 fix mixup stacktracereader

### PR DESCRIPTION
**Pull request by fpinel-microej**

### CODE REVIEW

File: ApplicationDeveloperGuide/stackTraceReader.rst [hunk 1]
```diff
@@ -44,25 +44,93 @@ This tool is a built-in platform tool.
 Use
 ===
 
+---------------------------
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/stackTraceReader.rst:47]: I suggest to merge the title

Use (Standalone Application)
===================

Use (Sandboxed Application)
===================

File: ApplicationDeveloperGuide/stackTraceReader.rst [hunk 2]
```diff
@@ -41,28 +41,91 @@ Installation
...
    :align: center
-   :width: 425px
-   :height: 124px
+   :scale: 100%
 
    Code to Dump a Stack Trace
 
+To be able to decode an application stack trace, the stack trace reader
+tool requires the application binary file 
+(in the case of a platform with full BSP connection (see :ref:`BSP Connection Cases <bsp_connection_cases>`), the file is ``application.out`` in the output folder).
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/stackTraceReader.rst:59]: avoid nested parenthesis

File: ApplicationDeveloperGuide/stackTraceReader.rst [hunk 3]
```diff
@@ -41,28 +41,91 @@ Installation
 This tool is a built-in platform tool.
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/stackTraceReader.rst:41]: - blur the platform / vd in the screeshots

File: ApplicationDeveloperGuide/stackTraceReader.rst [hunk 4]
```diff
@@ -41,28 +41,91 @@ Installation
...
+   :alt: Select Stack Trace Reader Tool
+   :align: center
+   :scale: 100%
+
+   Select Stack Trace Reader Tool
+
+In :guilabel:`Configuration` tab, browse the previously generated application
+binary file with debug information (``application.out`` in case of a Standalone Application with full BSP connection)
+
+.. figure:: images/STR_configuration_standalone.png
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/stackTraceReader.rst:93]: - remove Serial To Socket transmitter launch in standalone screenshot

File: ApplicationDeveloperGuide/stackTraceReader.rst [hunk 5]
```diff
@@ -41,28 +41,91 @@ Installation
...
+project and click on :guilabel:`Run As...` > :guilabel:`Run Configurations...`.
+
+Create a new MicroEJ Tool configuration. In the :guilabel:`Execution` tab, select your target platform, then select the :guilabel:`Stack Trace Reader` tool. Set an output folder in the :guilabel:`Output folder` field.
+
+.. figure:: images/STR_selecttool_standalone.png
+   :alt: Select Stack Trace Reader Tool
+   :align: center
+   :scale: 100%
+
+   Select Stack Trace Reader Tool
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/stackTraceReader.rst:88]: Stack Trace Reader Tool Configuration (Platform Selection)

File: ApplicationDeveloperGuide/stackTraceReader.rst [hunk 6]
```diff
@@ -75,88 +138,55 @@ without debug information).
...
 
-.. figure:: images/STR_selecttool.png
+.. figure:: images/STR_selecttool_sandboxed.png
    :alt: Select Stack Trace Reader Tool
    :align: center
-   :width: 926px
-   :height: 306px
+   :scale: 100%
 
    Select Stack Trace Reader Tool
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/stackTraceReader.rst:165]: Stack Trace Reader Tool Configuration (Virtual Device Selection)

File: ApplicationDeveloperGuide/stackTraceReader.rst [hunk 7]
```diff
@@ -75,88 +138,55 @@ without debug information).
...
-
 Click on :guilabel:`Run` button and copy/paste the trace into the Eclipse
 console. The decoded trace is dumped and the line corresponding to the
 application hook is now readable.
 
-.. TODO this figure was controlled with a conditional, isSandbox
 
-.. figure:: images/STR_console.png
+.. figure:: images/STR_console_sandboxed.png
    :alt: Read the Stack Trace
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/stackTraceReader.rst:184]: Stack Trace Reader Console

File: ApplicationDeveloperGuide/stackTraceReader.rst [hunk 8]
```diff
@@ -41,28 +41,91 @@ Installation
...
+Click on :guilabel:`Run` button and copy/paste the trace into the Eclipse
+console. The decoded trace is dumped and the line corresponding to the
+application hook is now readable.
+
+.. figure:: images/STR_trace-read.png
+   :alt: Read the Stack Trace
+   :scale: 100%
+   :align: center
+   
+   Read the Stack Trace
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/stackTraceReader.rst:109]: Stack Trace Reader Console

File: ApplicationDeveloperGuide/stackTraceReader.rst [hunk 9]
```diff
@@ -41,132 +41,164 @@ Installation
...
-   :width: 425px
-   :height: 124px
+   :scale: 100%
 
    Code to Dump a Stack Trace
 
-Write a new line to dump the currently executed stack trace on the
+To be able to decode an application stack trace, the stack trace reader
+tool requires the application binary file 
+(in the case of a platform with full BSP connection (see :ref:`BSP Connection Cases <bsp_connection_cases>`), the file is ``application.out`` in the output folder).
```
Comment (jerome-leroux) [inline ApplicationDeveloperGuide/stackTraceReader.rst:59]:
```suggestion
To decode an application stack trace, the stack trace reader tool requires the application executable ELF file.
In the case of a platform with full BSP connection (see :ref:`BSP Connection Cases <bsp_connection_cases>`), the file is ``application.out`` in the output folder. 
In the other cases, the ELF file is generated by the C toolchain when building the BSP project (usually a ``.out`` or ``.axf`` file).
```

**Review by jerome-leroux — CHANGES_REQUESTED** [review]

File: ApplicationDeveloperGuide/stackTraceReader.rst [hunk 10]
```diff
@@ -41,132 +41,164 @@ Installation
...
-       :width: 80.0%
-       :align: center
-       
-       Stack Trace Output
-
 To create a new MicroEJ Tool configuration, right-click on the application
 project and click on :guilabel:`Run As...` > :guilabel:`Run Configurations...`.
 
-In :guilabel:`Execution` tab, select the :guilabel:`Stack Trace Reader` tool.
+Create a new MicroEJ Tool configuration. In the :guilabel:`Execution` tab, select your target platform, then select the :guilabel:`Stack Trace Reader` tool. Set an output folder in the :guilabel:`Output folder` field.
```
[the remark this answers is not in the document]
Comment (frivieremicroej) [reply ApplicationDeveloperGuide/stackTraceReader.rst:158]: Felix tested with only the `.fodbg` file.

(the Kernel executable may not be able to the app developer)

We should explain both cases: (the base option is mandatory)
- If you have the Kernel executable => put it in the base option and put the `.fodbg` in additionnal files
- If you don't have the Kernel executable => put the `.fodbg` in the base option

File: ApplicationDeveloperGuide/stackTraceReader.rst [hunk 11]
```diff
@@ -41,132 +41,164 @@ Installation
...
+   :align: center
+   
+   Read the Stack Trace
+
+The stack trace reader can simultaneously decode heterogeneous stack
+traces with lines owned by different applications and the firmware.
+Other debug information files can be appended using the
+:guilabel:`Additional object files` option. Lines owned by the firmware can be
+decoded with the firmware debug information file (optionally made
+available by your firmware provider).
```
[the remark this answers is not in the document]
Comment (frivieremicroej) [reply ApplicationDeveloperGuide/stackTraceReader.rst:116]: ok see comment below => to be moved to the Use (Sandboxed Application) section

File: ApplicationDeveloperGuide/stackTraceReader.rst (hunk 10 above)
[the remark this answers is not in the document]
Comment (fpinel-microej) [reply ApplicationDeveloperGuide/stackTraceReader.rst:152]: I am not sure what is the difference here, how can I (or any developer for that matter) get access to the Kernel executable and use it with the stack trace reader ? From the previous version of the tutorial I gathered that only the `.fodbg` file was needed to use the STR, not sure how to proceed from here.

[the remark this answers is not in the document]
Comment (fpinel-microej) [reply ApplicationDeveloperGuide/stackTraceReader.rst:152]: Following jleroux's explanations, wrote a first draft for the Kernel executable file usage. Feedback welcome

[the remark this answers is not in the document]
Comment (jerome-leroux) [reply ApplicationDeveloperGuide/stackTraceReader.rst:152]: Looks good to me

[the remark this answers is not in the document]
Comment (frivieremicroej) [reply ApplicationDeveloperGuide/stackTraceReader.rst:158]: Yes it's very clear.
I just pulled up the "multi-trace" capability in functional description.
Thanks @fpinel-microej

**Pull request merged**